examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,726 @@
+
+We train these embeddings in a simple linear model through backprobagation.
+After the embedding encoder is trained, we used it as a preprocessor to the
+input features of a Gradient Boosted Tree model.
+
+Note that the embeddings and a decision forest model cannot be trained
+synergically in one phase, since decision forest models do not train with backpropagation.
+Rather, embeddings has to be trained in an initial phase,
+and then used as static inputs to the decision forest model.
+"""
+
+"""
+### Implement feature encoding with embeddings
+"""
+
+
+def create_embedding_encoder():
+    inputs = create_model_inputs()
+    encoded_features = []
+    for feature_name in inputs:
+        if feature_name in CATEGORICAL_FEATURES_WITH_VOCABULARY:
+            vocabulary = CATEGORICAL_FEATURES_WITH_VOCABULARY[feature_name]
+            # Create a lookup to convert a string values to an integer indices.
+            # Since we are not using a mask token nor expecting any out of vocabulary
+            # (oov) token, we set mask_token to None and  num_oov_indices to 0.
+            lookup = layers.StringLookup(
+                vocabulary=vocabulary, mask_token=None, num_oov_indices=0
+            )
+            # Convert the string input values into integer indices.
+            value_index = lookup(inputs[feature_name])
+            # Create an embedding layer with the specified dimensions
+            vocabulary_size = len(vocabulary)
+            embedding_size = int(math.sqrt(vocabulary_size))
+            feature_encoder = layers.Embedding(
+                input_dim=len(vocabulary), output_dim=embedding_size
+            )
+            # Convert the index values to embedding representations.
+            encoded_feature = feature_encoder(value_index)
+        else:
+            # Expand the dimensions of the numerical input feature and use it as-is.
+            encoded_feature = tf.expand_dims(inputs[feature_name], -1)
+        # Add the encodded feature to the list.
+        encoded_features.append(encoded_feature)
+    # Concatenate all the encoded features.
+    encoded_features = layers.concatenate(encoded_features, axis=1)
+    # Create and return a Keras model with encoded features as outputs.
+    return keras.Model(inputs=inputs, outputs=encoded_features)
+
+
+"""
+### Build a linear model to train the embeddings
+"""
+
+
+def create_linear_model(encoder):
+    inputs = create_model_inputs()
+    linear_output = layers.Dense(units=1, activation="sigmoid")(encoder(inputs))
+
+    linear_model = keras.Model(inputs=inputs, outputs=linear_output)
+    linear_model.compile(
+        optimizer=keras.optimizers.Adam(),
+        loss=keras.losses.BinaryCrossentropy(),
+        metrics=[keras.metrics.BinaryAccuracy("accuracy")],
+    )
+    return linear_model
+
+
+embedding_encoder = create_embedding_encoder()
+history = run_experiment(
+    create_linear_model(embedding_encoder),
+    train_data,
+    test_data,
+    num_epochs=3,
```

Comment:
The number of epoch only make sense to the linear or embedding layer.

Reply:
Yes. That is why I am only passing num_epochs when training the linear model, not when training a DF model (default is 1). Am I missing something?

I added the following text to clarify that in the "train and evaluate the model" step of the first experiment:

> Note that when training a Decision Forests model, only one epoch is needed to
> read the full dataset. Any extra steps will result in unnecessary slower training.
> Therefore, the default `num_epochs=1` is used in the `run_experiment` method.

Reply:
I see. So the code is equivalent to the following one, and batch_size=None will be ignored by the Keras API.

```python
num_epochs = 1
batch_size = None

model = tfdf.keras.RandomForest()
model.fit(train_dataset, epochs=num_epochs, batch_size=batch_size)
```

So the code is correct. It just feel a bit weird since specifying the number of epoch does not make sense for the TF-DF model, and readers might miss that part i.e. they might try to run the num_epoch.

Nit: What about having a **fit_args argument to the `run_experiment` function?

Reply:
Sounds good.